pandas/core/indexes/multi.py

```diff
@@ -2351,7 +2351,9 @@ def get_indexer(self, target, method=None, limit=None, tolerance=None):
                 raise NotImplementedError(
                     "tolerance not implemented yet for MultiIndex"
                 )
-            indexer = self._engine.get_indexer(target, method, limit)
+            indexer = self._engine.__class__.get_indexer_and_fill(
+                self.values, target, method=method, limit=limit
```

Comment:
I thought it would be preferable (per @jbrockmendel's point about keeping `Index` objects out of the cython code) to just pass the MultiIndex object's tuples directly to the cython code